app/assets/javascripts/discourse/app/controllers/user-badges.js

```diff
@@ -1,14 +1,27 @@
 import Controller, { inject as controller } from "@ember/controller";
-import { alias, sort } from "@ember/object/computed";
+import { action, computed } from "@ember/object";
+import { alias, filterBy, sort } from "@ember/object/computed";
 
 export default Controller.extend({
   user: controller(),
   username: alias("user.model.username_lower"),
   sortedBadges: sort("model", "badgeSortOrder"),
+  favoriteBadges: filterBy("model", "is_favorite", true),
+  canFavorite: computed(
+    "favoriteBadges.length",
+    "model.meta.max_favorites",
+    function () {
+      return this.favoriteBadges.length < this.model.meta.max_favorites;
+    }
+  ),
```

Comment:
Ember has a "less than" (called `lt`) helper that can do this for you 😉 

```suggestion
  canFavorite: lt("favoriteBadges.length", "model.meta.max_favorites"),
```

Reply:
Ah, so I tried using that. However, it looks like Ember's `lt` helper expects the second value to be constant so I couldn't reference `model`.

Reply:
Right, forgot about that part 👍